enterprise/server/ociregistry/ociregistry_test.go

```diff
@@ -47,8 +51,31 @@ func runMirrorRegistry(t *testing.T, env environment.Env, counter *atomic.Int32)
 	return listenAddr
 }
 
+func runByteStreamServer(ctx context.Context, t *testing.T, env *testenv.TestEnv) *grpc.ClientConn {
```

Comment:
I've seen this code (or code like it) sprinkled throughout the tests. Happy to pull something out into `testutil`.

Reply:
I think we should be able to setup the byte stream server automatically as part of `GetTestEnv`.

Reply:
For sure. I started down this road of copying `runByteStreamServer` code into `GetTestEnv`, but I have no conception of what the right interface is. Are you cool with me landing this PR once other feedback is addressed, and then modifying `GetTestEnv` in a separate change?

Reply:
Yeah it make sense for that to be a separate change. I think it may require changes to more files than what's already in this PR.

Reply:
FYI there is a `testcache` package which should have this logic; not all tests have been migrated to it though. Personally I would prefer to avoid adding more things to GetTestEnv as it will reduce our cache hit rate for tests, but don't feel strongly about it